src/fides/api/models/sql_models.py

```diff
@@ -493,6 +559,27 @@ def purpose(cls) -> Case:
             else_=None,
         )
 
+    @property
+    def undeclared_data_categories(self) -> Set[str]:
+        """
+        Aggregates a unique set of data categories across the collections in the associated datasets and
+        returns the data categories that are not defined directly on this or any sibling privacy declarations.
+        """
+
+        # all data categories from the datasets
+        dataset_data_categories = set()
+        for dataset in self.datasets:
+            dataset_data_categories.update(dataset.field_data_categories)
+
+        # all data categories specified directly on this and sibling privacy declarations
+        declared_data_categories = set()
+        for privacy_declaration in self.system.privacy_declarations:
```

Comment:
I'm a little confused about the organization here.  So each privacy declaration has undeclared data categories that are looking at itself but also all of the other privacy declarations? Should this property be at this level? (instead a property on the system, maybe, privacy_declaration_undeclared_data_categories?

Reply:
Ah yes, there's a tiny detail here which is that the `undeclared_data_categories` on the `PrivacyDeclaration` report on the data categories tied to the `datasets` on the `PrivacyDeclaration`. So the search space is all of the data categories across all PDs but for each PD we're only evaluating the datasets associated with itself.

Reply:
ah ok can you make this a code comment?